File: Anul-3/Semestrul-I/PPD/Tema-1/Java/src/main/java/com/example/data/MatrixGenerator.java
package com.example.data;

import java.io.File;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.Random;

public class MatrixGenerator {

    public static Random rand = new Random();
    private final String matrixFile;
    private final String filterFile;
    private final int rows;
    private final int cols;
    private final int k;

    public MatrixGenerator(String matrixFile, String filterFile,int N,int M, int k) {
        this.matrixFile = matrixFile;
        this.filterFile = filterFile;
        this.rows = N;
        this.cols = M;
        this.k = k;
    }

    public static void writeMatrix(int[][] matrix, String filename) {
        try (PrintWriter out = new PrintWriter(filename)) {
            out.println(matrix.length+" "+matrix[0].length);
            for (int i = 0; i < matrix.length; i++) {
                for (int j = 0; j < matrix[i].length; j++) {
                    out.print(matrix[i][j] + " ");
                }
                out.println();
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private void generateMatrix(int rows, int cols) {
        int[][] matrix = new int[rows][cols];
        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                matrix[i][j] = rand.nextInt(100);
            }
        }
        writeMatrix(matrix, matrixFile);
    }

    private void generateFilter(int k) {
        int[][] matrix = new int[k][k];
        for (int i = 0; i < k; i++) {
            for (int j = 0; j < k; j++) {
                matrix[i][j] = rand.nextInt(10);
            }
        }
        writeMatrix(matrix, filterFile);
    }

    public void run(){
        generateMatrix(rows, cols);
        generateFilter(k);
    }
}

File: Anul-3/Semestrul-I/PPD/Tema-2/C++/source/MatrixGenerator.cpp
//
// Created by vladb on 18/10/2025.
//

#include "../header/MatrixGenerator.h"
#include <string>
#include <fstream>
#include <chrono>
using namespace std;

MatrixGenerator::MatrixGenerator(const string &matrix_file, const string &filter_file, int n, int m, int k)
    : matrix_file(matrix_file), filter_file(filter_file), rows(n), cols(m), k(k) {}


void MatrixGenerator::generateMatrix() const {
    ofstream outM(matrix_file);
    outM << rows << " " << cols<<endl;
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            outM << rand() % 20 << " ";
        }
        outM << endl;
    }




}
void MatrixGenerator::generateFilter() const {
    ofstream outFilter(filter_file);
    outFilter << k << " " << k<<endl;
    for (int i = 0; i < k; i++) {
        for (int j = 0; j < k; j++) {
            outFilter << rand() % 100 << " ";
        }
        outFilter << endl;
    }
}
void MatrixGenerator::writeMatrixToFile(int **matrix, const string &matrix_file,const int rows,const int cols) {
    ofstream outM(matrix_file);
    for (int i=0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            outM << matrix[i][j] << " ";
        }
        outM << endl;
    }
}

